### PR TITLE
[IE C Samples] Implement bmp reader

### PR DESCRIPTION
**Pull request by dpigasin**

### CODE REVIEW

File: samples/c/common/opencv_c_wrapper/bmp_reader.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+typedef struct BitMapType {
+    BmpHeader header;
+    BmpInfoHeader infoHeader;
+    int width, height;
+    unsigned char* data;
+} BitMap;
+
+
+int readBmpImage(const char* fileName, BitMap* image);
```
Comment (ilya-lavrenov) [inline samples/c/common/opencv_c_wrapper/bmp_reader.h:33]: is it possible not to copy code, but `format_reader_c` will just wrap `format_reader`?

**Review by vladimir-dudnik — APPROVED** [review]

File: samples/c/common/opencv_c_wrapper/bmp_reader.h (hunk 1 above)
Comment (vladimir-dudnik) [reply samples/c/common/opencv_c_wrapper/bmp_reader.h:33]: @ilya-lavrenov do you mean creating common folder under sample subfolder to be shared between c/cpp samples ?

Comment (ilya-lavrenov) [reply samples/c/common/opencv_c_wrapper/bmp_reader.h:33]: or we can install this common part to both C and C++ places like we do for CMakeLists.txt https://github.com/openvinotoolkit/openvino/blob/master/samples/CMakeLists.txt#L61-L63

**Review by ilya-lavrenov — APPROVED** [review]

File: samples/c/common/opencv_c_wrapper/bmp_reader.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    int padSize = image->width & 3;
+    char pad[3];
+    size_t size = image->width * image->height * 3;
+
+    image->data = malloc(sizeof(char) * size);
+
+    fseek(input, image->header.offset, 0);
+
+    // reading by rows in invert vertically
+    for (int i = 0; i < image->height; i++) {
```
Comment (vladimir-dudnik) [inline samples/c/common/opencv_c_wrapper/bmp_reader.c:49]:
```suggestion
    int i;
    for (i = 0; i < image->height; i++) {
```

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
please fix compilation on centos (too old gcc)
21:38:15  [2021-12-08 18:38:06,382] [140516464346880] build ie ERROR: /home/jenkins/agent/workspace/private-ci/ie/build-linux-centos76/b/repos/openvino/samples/c/common/opencv_c_wrapper/bmp_reader.c:49:5: error: ‘for’ loop initial declarations are only allowed in C99 mode

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**